android/Drawer/app/src/main/java/com/example/drawer/CanvasGrid.java

```diff
@@ -108,10 +151,20 @@ protected void onDraw(Canvas canvas) {
                     canvas.drawRect(i * cellLength, j * cellLength,
                             (i + 1) * cellLength, (j + 1) * cellLength,
                             blackPaint);
+//                    if (!pointQueue.contains(new Point(i, j))) {
+//                        pointQueue.add(new Point(i, j));
+//                    }
+                }
+                if (pureCellChecked[i][j]) {
+                    if (!pointQueue.contains(new Point(i, j))) {
+                        pointQueue.add(new Point(i, j));
+                    }
                 }
             }
         }
 
+        // TODO Are getters needed here? Perhaps attributes should be used -KC
```

Comment:
Good question, are they needed? I see some getters lines 63-77, however, since activity classes don't interact with each other, it makes more sense to simply leave them as attributes, no?